### PR TITLE
[Sdk 289] Push dependecies removel crash

### PR DESCRIPTION
**Pull request by ZahidZafar**

### CODE REVIEW

File: Assets/Plugins/Notifications/Impls/Android/AndroidNotificationsService.cs [hunk 1]
```diff
@@ -34,9 +35,13 @@ internal AndroidNotificationsService(Transform countlyGameObject, CountlyConfigu
...
             _bridge.Log = Log;
 
-            AndroidJavaClass countlyPushPlugin = new AndroidJavaClass(CountlyPushPluginPackageName);
-            countlyPushPlugin.CallStatic("setEnableLog", config.EnableConsoleLogging);
-
+            try {
+                AndroidJavaClass countlyPushPlugin = new AndroidJavaClass(CountlyPushPluginPackageName);
+                countlyPushPlugin.CallStatic("setEnableLog", config.EnableConsoleLogging);
+            } catch (Exception ex) {
+                Log.Error("[AndroidNotificationsService] Exception: " + ex.Message);
```
Comment (ArtursKadikis) [inline Assets/Plugins/Notifications/Impls/Android/AndroidNotificationsService.cs:42]: Add more context to the error message

File: Assets/Plugins/Notifications/Impls/Android/AndroidNotificationsService.cs [hunk 2]
```diff
@@ -34,9 +35,13 @@ internal AndroidNotificationsService(Transform countlyGameObject, CountlyConfigu
...
+                throw new FileNotFoundException(ex.Message);
```
Comment (ArtursKadikis) [inline Assets/Plugins/Notifications/Impls/Android/AndroidNotificationsService.cs:43]: This exception should never be thrown, why does this call exist here?

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

File: Assets/Plugins/Notifications/Impls/Android/AndroidNotificationsService.cs (hunk 2 above)
Comment (ZahidZafar) [reply Assets/Plugins/Notifications/Impls/Android/AndroidNotificationsService.cs:43]: on the base of this exception, ProxyNotificationServcie don't keep it's instance and  block all future request make to it.

                countlyPushPlugin.CallStatic("setEnableLog", config.EnableConsoleLogging);
                
                this code enable/disable the log in native android code.

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Failure to initialise a internal component should be notified in a better way than throwing an exception.

File: Assets/Example/Scripts/CountlyEntryPoint.cs [hunk 3]
```diff
@@ -38,7 +38,7 @@ private void Awake()
         countly = Countly.Instance;
     }
 
-    private void OnDisable()
+    private void OnDetroy()
```
Comment (ArtursKadikis) [inline Assets/Example/Scripts/CountlyEntryPoint.cs:41]: mistake

File: Assets/Plugins/Notifications/Impls/ProxyNotificationsService.cs [hunk 4]
```diff
@@ -13,20 +15,30 @@ public class ProxyNotificationsService : INotificationsService
         private readonly CountlyLogHelper _logHelper;
         private readonly Transform _countlyGameObject;
         private readonly INotificationsService _service;
+        public bool IsInitializedWithoutError { get; set; }
 
         internal ProxyNotificationsService(Transform countlyGameObject, CountlyConfiguration config, CountlyLogHelper logHelper, Action<IEnumerator> startCoroutine, EventCountlyService eventCountlyService)
         {
             _logHelper = logHelper;
-            _logHelper.Debug("[ViewCProxyNotificationsServiceountlyService] Initializing.");
+            _logHelper.Debug("[ProxyNotificationsServiceountlyService] Initializing.");
```
Comment (ArtursKadikis) [inline Assets/Plugins/Notifications/Impls/ProxyNotificationsService.cs:23]: this name is wrong and should be improved

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]

File: Assets/Example/Scripts/CountlyEntryPoint.cs (hunk 3 above)
Comment (ZahidZafar) [reply Assets/Example/Scripts/CountlyEntryPoint.cs:41]: it's not. listener will get remove on destroy not on disable. Disable could call many times, i.e when we disable game object etc...

**Review by ArtursKadikis — APPROVED** [review]

**Pull request merged**